scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/macros/impl/WriteSupportProvider.scala

```diff
@@ -0,0 +1,106 @@
+package com.twitter.scalding.parquet.tuple.macros.impl
+
+import com.twitter.bijection.macros.impl.IsCaseClassImpl
+import parquet.io.api.RecordConsumer
+import parquet.schema.MessageType
+
+import scala.reflect.macros.Context
+
+object WriteSupportProvider {
+
+  def toWriteSupportImpl[T](ctx: Context)(implicit T: ctx.WeakTypeTag[T]): ctx.Expr[(T, RecordConsumer, MessageType) => Unit] = {
+    import ctx.universe._
+
+    if (!IsCaseClassImpl.isCaseClassType(ctx)(T.tpe))
+      ctx.abort(ctx.enclosingPosition,
+        s"""We cannot enforce ${T.tpe} is a case class,
+            either it is not a case class or this macro call is possibly enclosed in a class.
+            This will mean the macro is operating on a non-resolved type.""")
+
+    def matchField(idx: Int, fieldType: Type, fValue: Tree, groupName: TermName): (Int, Tree) = {
+      def writePrimitiveField(wTree: Tree) =
+        (idx + 1, q"""rc.startField($groupName.getFieldName($idx), $idx)
+                      $wTree
+                      rc.endField($groupName.getFieldName($idx), $idx)""")
+
+      def writeGroupField(subTree: Tree) =
+        q"""rc.startGroup()
+            rc.startField($groupName.getFieldName($idx), $idx)
+            $subTree
+            rc.endField($groupName.getFieldName($idx), $idx)
+            rc.endGroup()
+         """
```

Comment:
Am I missing something or should the startField/endField calls be around the startGroup/endGroup and not the other way around?